doc/src/vpr/debug_aids.rst

```diff
@@ -24,3 +24,77 @@ Finally, if you need to route an FPGA whose routing architecture cannot be descr
+
+Placer and Router Debugger
+==========================
+
+.. image:: https://github.com/verilog-to-routing/verilog-to-routing.github.io/blob/master/img/debuggerWindow.png
+    :align: center
+
+Adding a breakpoint
+~~~~~~~~~~~~~~~~~~~
+
+Currently the user is required to have graphics on in order to set breakpoints. By clocking the “Debug” button, the debugger window opens up and from there the user can enter integer values in the entry fields and set breakpoints. A more advanced option is using expressions which allows a wider variety of settings since the user can incorporate multiple variables and use boolean operators. This option is found by clicking the “Advanced” button in the debugger window. Using an expression is more accurate than the entry fields when setting multiple breakpoints.
+
+Enabling/Disabling a breakpoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Enabling and disabling breakpoints are done using the checkboxes in front of each breakpoint in the breakpoint list. The breakpoint is enabled when the box is checked and disabled otherwise.
+
+Deleting a breakpoint
+~~~~~~~~~~~~~~~~~~~~~
+
+Deleting a breakpoint is done using the trash can button in front of each breakpoint in the breakpoint list.
+
+Reaching a breakpoint
+~~~~~~~~~~~~~~~~~~~~~
+
+Upon reaching a breakpoint, the program will stop, notify the user which breakpoint was encountered, and give a summary of the current variable values. This information is presented through a pop-up window and printed to the terminal as well.
+
+Available Variables
+~~~~~~~~~~~~~~~~~~~
+
+.. image:: https://github.com/verilog-to-routing/verilog-to-routing.github.io/blob/master/img/advancedWindow.png
+    :align: center
+
+You can also find the variables’ list in the Advanced Settings Window, on the left.
+
+**Placer Variables**
+
+  * move_num: every placer iteration counts as a move, so the user can stop the program after a certain number of moves. This breakpoint can be enabled through the entry field on the main debugger window or using an expression. It should be noted however, that using the entry field would proceed the specified number of moves. (as in the second example)
+          * Ex. move _num == 33
+          * Ex. move_num += 4
+  * temp_count: every time the temperature is updated it counts as an increase to temp_count. This breakpoint can be enabled through the entry field on the main debugger window or using an expression. It should be noted however, that using the entry field would proceed the specified number of temperatures. (as in the second example)
+          * Ex. temp_count == 5
+          * Ex. temp_count += 5
+  * from_block:  in every placer move one or more blocks are relocated. From_block specifies the first block that is relocated in every move. This breakpoint can be enabled through the entry field on the main debugger window or using an expression.
```

Comment:
Do the from_block and in_blocks_affected breakpoints stop before the blocks are moved or after?

Reply:
All changes have been made :)